### PR TITLE
fix: Accept either database_name or namespace_name

### PR DESCRIPTION
**Pull request by carols10cents**
Connects to https://github.com/influxdata/idpe/issues/16465

As far as I can tell, this is only a problem when using JSON -- I did not change the field in the proto in https://github.com/influxdata/influxdb_iox/pull/6127.

### CODE REVIEW

**Review by jacobmarble — CHANGES_REQUESTED** [review]
Thanks for this. Let's give a chance to the IDPE change. If it doesn't work, then we'll merge this.

File: service_grpc_flight/src/lib.rs [hunk 1]
```diff
@@ -127,22 +132,60 @@ struct ReadInfo {
...
+            sql_query: read_info.sql_query,
+        })
+    }
+}
+
+#[derive(Deserialize, Debug, Clone)]
+/// Body of the `Ticket` serialized and sent to the do_get endpoint.
+pub struct JsonReadInfo {
+    namespace_name: Option<String>,
+    // Deprecated; use `namespace_name` instead. This is needed until all clients are updated.
```
Comment (alamb) [inline service_grpc_flight/src/lib.rs:150]: 👍

**Review by alamb — APPROVED** [review]
The code looks great -- thank you @carols10cents 

I like that we have this as a backup if the additional updates don't work

**Comment by alamb** [comment]
@jacobmarble  since the IDPE PR seems to be having issues, I would like to propose we merge this one asap to unbreak prod sql queries ASAP. Is that ok?

**Review by jacobmarble — APPROVED** [review]

**Comment by alamb** [comment]
https://github.com/influxdata/idpe/pull/16466 appears to have fixed our customer issues. I think this PR is still good to have ready to go in case we see any other issues.

Thank you again for the quick work @carols10cents  🙏

**Comment by carols10cents** [comment]
Thank you to @jacobmarble for cleaning up after the mess I made!! I'll close this PR but leave the branch around just in case.

**Pull request closed without merging**